Makefile.am

```diff
@@ -168,6 +167,20 @@ etags:
 	cd $(srcdir)/tools/ir-generator; ctags -e -R --langmap=Flex:+.l,YACC:+.ypp \
 	    . ../../lib
 
+# automake's default "make check" rule runs into "Argument list too long" errors on
+# Linux when TESTS gets to big, because of the peculiar way in which it recursively
+# invokes make with assignments on the command line.  These in turn end up in the
+# MAKEFLAGS environment variable which causes the problem when invoking bash, as
+# it exceeds Linux's 128K limit on a single environment variable.  So we override
+# with a simpler recipe that doesn't do that.  This gives us warnings from make
+# about multiple recipes for 'check', but the right one (this one) ends up being used
+check: all-am $(BUILT_SOURCES) $(check_LTLIBRARIES) $(check_PROGRAMS) cpplint-quiet
```

Comment:
So I cleaned this up by changing it to override check-am instead of check (the default check just depends on `$(BUILT_SOURCES)`, then does a `make check-am`.)  Hopefully that won't break horribly on other versions of automake.